next-app/lib/headersConstruct.js

```diff
@@ -1,6 +1,8 @@
-var fetchDataFunc = function () {
-    let apiKey = localStorage.getItem('apiKey');
-    let host = localStorage.getItem('hostKey');
+var fetchDataFunc = async function () {
+    let storedKeyPairs = await chrome.storage.local.get('keyPairs');
+    let apiKey = storedKeyPairs["keyPairs"][0]["apiKey"];
+    let host = storedKeyPairs["keyPairs"][0]["hostKey"];
```

Comment:
we should remake it to a loop, because we will have multiple key pairs, meaning we cant just take [0]. Either we change it now or we have to write it down somewhere so we don't forget